src/components/textbox/textbox.test.js

```diff
@@ -0,0 +1,712 @@
+
+    it("should render Textbox with readOnly prop", () => {
+      CypressMountWithProviders(<TextboxComponent readOnly />);
+
+      textboxInput().and("have.attr", "readOnly");
+      textbox().should("have.attr", "readOnly");
+    });
+
+    it.each(["error", "warning", "info"])(
+      "should verify Textbox is displayed with %s validation icon on input",
+      (type) => {
+        CypressMountWithProviders(
+          <TextboxComponent
+            labelInline
+            labelAlign="right"
+            {...{ [type]: "Message" }}
+          />
+        );
+
+        textbox()
+          .children(1)
+          .children()
+          .children()
+          .should("have.attr", "data-component", "icon")
+          .and("have.attr", "data-element", type);
+      }
+    );
+
+    it.each(["error", "warning", "info"])(
+      "should verify Textbox is displayed with %s validation icon on label",
+      (type) => {
+        CypressMountWithProviders(
+          <TextboxComponent
+            labelInline
+            labelAlign="right"
+            validationOnLabel
+            {...{ [type]: "Message" }}
+          />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .children(1)
+          .children()
+          .children()
+          .should("have.attr", "data-component", "icon")
+          .and("have.attr", "data-element", type);
+      }
+    );
+
+    it.each([
+      ["rgb(203, 55, 74)", "error", true],
+      ["rgb(239, 103, 0)", "warning", true],
+      ["rgb(0, 96, 167)", "info", true],
+      ["rgb(102, 132, 148)", "error", false],
+      ["rgb(102, 132, 148)", "warning", false],
+      ["rgb(102, 132, 148)", "info", false],
+    ])(
+      "should verify Textbox input border colour is %s when validation is %s and boolean prop is %s",
+      (borderColor, type, bool) => {
+        CypressMountWithProviders(
+          <TextboxComponent
+            labelInline
+            labelAlign="right"
+            {...{ [type]: bool }}
+          />
+        );
+
+        textbox().should("have.css", "border-bottom-color", borderColor);
+      }
+    );
+
+    it("should render Textbox with leftChildren prop", () => {
+      CypressMountWithProviders(
+        <TextboxComponent leftChildren={<Button>Test</Button>} />
+      );
+
+      textbox()
+        .children(0)
+        .should("have.attr", "data-component", "button")
+        .and("be.visible");
+    });
+
+    it.each([CHARACTERS.SPECIALCHARACTERS, CHARACTERS.DIACRITICS])(
+      "should render Textbox with prefix prop set to %s",
+      (prefix) => {
+        CypressMountWithProviders(<TextboxComponent prefix={prefix} />);
+
+        textboxPrefix()
+          .should("have.text", prefix)
+          .and("have.css", "font-size", "16px")
+          .and("have.css", "font-weight", "900")
+          .and("have.css", "margin-right", "8px");
+      }
+    );
+
+    it.each([
+      [true, "input", "label"],
+      [false, "label", "input"],
+    ])(
```

Comment:
unfortunately, these tests don't work as intended - if you switch the order of "input" and "label" in one of them, which should make it fail, it still passes. I've not yet been able to figure out how to fix it - happy to leave it to you as you know Cypress much better than me! - but these needs fixing otherwise this test has no value.

Reply:
thank you for this comment, fixed this as well